php/WC_Gateway_Blockonomics.php

```diff
@@ -122,17 +452,24 @@ public function handle_requests()
         $txid = isset($_GET['txid']) ? sanitize_text_field(wp_unslash($_GET['txid'])) : "";
         $rbf = isset($_GET['rbf']) ? wp_validate_boolean(intval(wp_unslash($_GET['rbf']))) : "";
 
+        $test_setup = isset($_GET["test_setup"]) ? sanitize_text_field(wp_unslash($_GET['test_setup'])) : "";
+        $api_key = isset($_GET["api_key"]) ? sanitize_text_field(wp_unslash($_GET['api_key'])) : "";
+        $btc_active = isset($_GET["btc_active"]) ? $_GET["btc_active"] : false;
+        $bch_active = isset($_GET["bch_active"]) ? $_GET["bch_active"] : false;
+
         include_once 'Blockonomics.php';
         $blockonomics = new Blockonomics;
 
         if ($finish_order) {
             $order_id = $blockonomics->decrypt_hash($finish_order);
             $blockonomics->redirect_finish_order($order_id);
-        }else if ($get_amount && $crypto) {
+        } else if ($get_amount && $crypto) {
             $order_id = $blockonomics->decrypt_hash($get_amount);
             $blockonomics->get_order_amount_info($order_id, $crypto);
-        }else if ($secret && $addr && isset($status) && $value && $txid) {
+        } else if ($secret && $addr && isset($status) && $value && $txid) {
             $blockonomics->process_callback($secret, $addr, $status, $value, $txid, $rbf);
+        } else if ($test_setup) {
+            $blockonomics->settings_test_setup($api_key, $btc_active, $bch_active);
```

Comment:
It will be better to use the `wp_ajax_*` hook to handle the test setup request. This will ensure only logged in users are making the request and will keep this functionality separate from the payment processing logic:
`add_action( 'wp_ajax_blockonomics_test_setup', 'blockonomics_test_setup' );`
We will still be able to perform the request in vanilla js using `fetch()`.